hash-table/src/hash_table.c

```diff
@@ -89,8 +85,9 @@ static void ht_resize(ht_hash_table* ht, int direction) {
     ht_hash_table* new_ht = ht_new_sized(new_size_index);
     // Iterate through existing hash table, add all items to new
     for (int i = 0; i < ht->size; i++) {
-        if (ht->items[i] != NULL && ht->items[i] != HT_DELETED_ITEM) {
-            ht_insert(new_ht, ht->items[i]->key, ht->items[i]->value);
+        ht_item *it = ht->items[i];
```

Comment:
I like to factor out common expressions like this manually.  Makes it easier to see what's going on, and easier to fix later when I find out I got it wrong.